### PR TITLE
feat: Node autoscaling and more

### PR DESCRIPTION
**Pull request by edepremare-cs**
This PR is related to https://github.com/COPRS/rs-issues/issues/357

Major features:
 - cluster infrastructure is now defined by node groups
 - expose rook-ceph dashboard on ceph.PLATFORM_DOMAIN_NAME
 - create a rs-infra-scaler component to scale the cluster
 - deploy the kubernetes cluster-autoscaler, the infra scaler, and safescaled into the cluster for node autoscaling
 - document node manual scaling, in sync with autoscaling

Minor features:
  - use grafana's loki output plugin for fluent-bit (better performance, security and stability)
  - refactor many playbooks

Improvements:
 - update rook-ceph operator (particularily for the daemonset priorities)
 - use kubespray etcd metrics service (better deployment)
 - add mongoDB root password (gives an administrator more flexibility)

Fixes:
 - fix https://github.com/COPRS/rs-issues/issues/363
 - fix https://github.com/COPRS/rs-issues/issues/375
 - remove tags for security cots, to allow downloading latest repo versions in future deployments (clamav, auditd)

### CODE REVIEW

File: apps/70-fluent-bit/values.yaml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 kind: DaemonSet
 
 image:
-  repository: fluent/fluent-bit
-  tag: 1.9.1
-  pullPolicy: IfNotPresent
+  repository: artifactory.coprs.esa-copernicus.eu/rs-docker/fluent-bit-plugin-loki
```
Comment (alambare-csgroup) [inline apps/70-fluent-bit/values.yaml:4]: could not we add this plugin with an initContainer / post-start script ?

File: roles/infra-defaults/defaults/main.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# Ssh keys directory path
+ssh_key_path: "{{ inventory_dir }}/artifacts/.ssh"
+# SafeScale operator name
+safescale_operator: safescale
+
+# LDAP config
+base_dn: dc=cluster,dc=local
+ldap_server_url: ldap://127.0.0.1:30389
+
+etcd_metrics_service_labels:
```
Comment (alambare-csgroup) [inline roles/infra-defaults/defaults/main.yaml:14]: why not set this in kubespray inventory file ?

File: cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    - never
+
+- name: Update hosts.yaml inventory file
+  hosts: localhost
+  roles:
+    - { role: "infra-defaults"}
+    - { role: "safescale/update-hosts" }
+  tags: update_hosts
+
+- name: Add providerID spec to nodes
```
Comment (alambare-csgroup) [inline cluster.yaml:64]: Maybe this task miss a tag `never` as we cannot run it before kubespray ?

File: roles/infra-defaults/defaults/main.yaml (hunk 2 above)
Comment (edepremare-cs) [reply roles/infra-defaults/defaults/main.yaml:14]: This setting should not need to be modified, so it is better in the defaults file, to not overcharge the kubespray inventory file

File: cluster.yaml (hunk 3 above)
Comment (edepremare-cs) [reply cluster.yaml:64]: Right !

File: apps/40-rook-ceph/values.yaml [hunk 4]
```diff
@@ -1,10 +1,6 @@
-# default values at https://github.com/rook/rook/blob/master/deploy/charts/rook-ceph/values.yaml
-
 image:
   repository: rook/ceph
-  tag: v1.7.7
-
-logLevel: WARNING
+  tag: v1.9.4
```
[the remark this answers is not in the document]
Comment (edepremare-cs) [reply apps/40-rook-ceph/values.yaml:3]: We usually leave the image tags in the values files, so the helm can be updated independently from the images

File: apps/70-fluent-bit/values.yaml (hunk 1 above)
Comment (edepremare-cs) [reply apps/70-fluent-bit/values.yaml:4]: It is not actually a plugin, but a image containing the plugin, it cannot be done as a init container in the latest versions

File: roles/security/clamav/defaults/main.yaml [hunk 5]
```diff
@@ -1,4 +0,0 @@
----
-clamav:
-  # Required
-  clamav_version: "0.103.5+dfsg-1~20.04.1"
```
[the remark this answers is not in the document]
Comment (edepremare-cs) [reply roles/security/clamav/defaults/main.yaml:4]: Done !

File: doc/how-to/Cluster configuration.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+## Mandatory node groups specifics
+
+The node groups `kube_control_plane`, `gateway` and `infra` are mandatory and cannot have volumes.
+
+The amount of nodes in the `kube_control_plane` and `gateway` node groups are tied the SafeScale cluster complexity, meaning:
+ - small: 1 gateway, 1 kube_control_plane, 1 infra
+ - normal: 2 gateways, 2 kube_control_plane, 3 infra
+ - large: 2 gateways, 3 kube_control_plane, 7 infra
+
+The minimum size if the `infra` node group must be superior to the size shipped with the SafeScale complexity.
```
[the remark this answers is not in the document]
Comment (edepremare-cs) [reply doc/how-to/Cluster configuration.md:33]: Wrote a better one I think.

File: doc/how-to/Cluster scaling.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
+# Expand a node group
+
+
+> Keep in mind that the only ansible variables actually used are the ones in the `generated_inventory_vars.yaml` file and you can directly edit this file, retrieve it from a vault, upload it to a vault etc, see the [Credentials.md](./Credentials.md)
+
+> You must have run the `generate_inventory.yaml` playbook on your machine already.
```
[the remark this answers is not in the document]
Comment (edepremare-cs) [reply doc/how-to/Cluster scaling.md:6]: Removed the redundant information.

File: doc/how-to/Cluster scaling.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+1. Drain and delete the nodes from the kubernetes cluster
+
+For each node, run:
+```shellsession
+ansible-playbook collections/kubespray/remove-node.yml -i inventory/mycluster/hosts.yaml -b -e skip_confirmation=yes -e reset_nodes=false -e node=NODE_TO_DELETE
+```
+
+2. Delete nodes from the cloud provider
+```shellsession
+ansible-playbook cluster.yaml -i inventory/mycluster/hosts.yaml -t delete,update_hosts -e nodes_to_delete=NODE_TO_DELETE1,NODE_TO_DELETE2,...
```
[the remark this answers is not in the document]
Comment (edepremare-cs) [reply doc/how-to/Cluster scaling.md:46]: Done, indeed it is clearer

File: roles/security/wazuh/tasks/role-agent.yaml [hunk 9]
```diff
@@ -29,8 +39,8 @@
 - name: Join Wazuh Manager
   shell:
     cmd: /var/ossec/bin/agent-auth -m {{ wazuh_manager_ip }}
-  no_log: True
-  when: "inventory_hostname not in hostvars[groups['kube_control_plane'][0]]['wazuh_nodes_list'] "
+  no_log: False
```
[the remark this answers is not in the document]
Comment (edepremare-cs) [reply roles/security/wazuh/tasks/role-agent.yaml:42]: My mistake from debugging, will fix

File: roles/security/clamav/defaults/main.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (edepremare-cs) [reply roles/security/clamav/defaults/main.yaml:4]: Done

File: doc/how-to/Cluster configuration.md [hunk 10]
```diff
@@ -30,7 +30,7 @@ The amount of nodes in the `kube_control_plane` and `gateway` node groups are ti
  - normal: 2 gateways, 2 kube_control_plane, 3 infra
  - large: 2 gateways, 3 kube_control_plane, 7 infra
 
-The minimum size if the `infra` node group must be superior to the size shipped with the SafeScale complexity.
+The minimum size if the below defined `infra` node group must be superior to the size of the `infra` node group defined by the SafeScale complexity above.
```
Comment (alambare-csgroup) [inline doc/how-to/Cluster configuration.md:33]: The minimum size if -> The minimum size **of**

**Review by alambare-csgroup — APPROVED** [review]

**Pull request merged**